test/rules/use_initializing_formals_when_possible.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_initializing_formals_when_possible`
+
+num cos(num theta) {
+  return 0;
+}
+
+num sin(num theta) {
+  return 0;
+}
+
+class BadPoint1 {
+  num x, y;
+  BadPoint1(num x, num y) {
+    this.x = x; // LINT
+    this.y = y; // LINT
+  }
+}
+
+class BadPoint2 {
+  num x, y;
+  BadPoint2(this.x, num y) {
+    this.y = y; // LINT
+  }
+}
+
+class BadPoint3 {
+  num x, y;
+  BadPoint3(num a, num b) {
+    x = a; // LINT
+    y = b; // LINT
+  }
+}
+
+class BadPoint4 {
+  num x;
+  num y;
+  BadPoint4(num a, num b) {
+    x = a; // LINT
+    y = b; // LINT
+  }
+}
+
+class BadPoint5 {
+  num x, y;
+  BadPoint5.lint1(num a, this.y)
+      : this.x = a, // LINT
+        super();
+  BadPoint5.lint2(num a, this.y)
+      : x = a, // LINT
+        super();
+}
+
+class GoodPoint1 {
+  num x, y;
+  GoodPoint1(this.x, this.y);
+}
+
+class GoodPoint2 {
+  String name;
+  GoodPoint2(num x, num y) {
+    this.x = x; // OK
+    this.y = y; // OK
+  }
+  set x(num x) {
+    name = 'My x value is $x';
+  }
+
+  set y(num y) {
+    name = 'My y value is $y';
+  }
+}
+
+class GoodPoint3 {
+  String name;
+  GoodPoint3(num a, num b) {
+    x = a; // OK
+    y = b; // OK
+  }
+  get x {
+    return 0;
+  }
+
+  set x(num x) {
+    name = 'My x value is $x';
+  }
+
+  set y(num y) {
+    name = 'My y value is $y';
+  }
+}
+
+class GoodPoint4 extends GoodPoint1 {
+  num r, theta;
+  GoodPoint4(num r, num theta) : super(r * cos(theta), r * sin(theta)) {
+    this.r = r; // OK
+    this.theta = theta; // OK
```

Comment:
In the first local version of the linter, we run it in the SDK and we founded some lints with that case happening, and we realized that this:

```
GoodPoint4(this.r, this.theta) : super(r * cos(theta), r * sin(theta));
```

Is not possible to do, so it was a false positive and we updated the rule